test/services/AddressLookupServiceSpec.scala

```diff
@@ -0,0 +1,158 @@
+        address = customerAddressMax,
+        userAnswers = emptyUserAnswers.copy(warehouseList = warehouseMap))
+
+       res.warehouseList mustBe AddedWarehouse
+    }
+
+    "add to the cache the address of a warehouse when a user returns from address lookup frontend with missing address lines" in {
+      val addressLookupState = Warehousedetails
+      val warehouseMap = Map("1" -> Warehouse(Some("super cola"), UkAddress(List("33 Rhes Priordy", "East London"), "E73 2RP")))
+      val AddedWarehouseMissingLines = Map("1" -> Warehouse(Some("super cola"), UkAddress(List("33 Rhes Priordy", "East London"), "E73 2RP")),"12" -> Warehouse(Some(organisation), UkAddress(List(addressLine1, addressLine2), postcode)))
+      val customerAddressMissingLines: AlfResponse = AlfResponse (
+        Some(organisation),
+        List(addressLine1, addressLine2),
+        Some(postcode),
+        Some(countryCode)
+      )
+
+      val res = service.addAddressUserAnswers(addressLookupState = addressLookupState,
+        address = customerAddressMissingLines,
+        userAnswers = emptyUserAnswers.copy(warehouseList = warehouseMap))
+
+      res.warehouseList mustBe AddedWarehouseMissingLines
+    }
+
+    "add to the cache the address of a warehouse when a user returns from address lookup frontend with full address lines" in {
+      val addressLookupState = Warehousedetails
+      val warehouseMap = Map("1" -> Warehouse(Some("super cola"), UkAddress(List("33 Rhes Priordy", "East London"), "E73 2RP")))
+
+      val res = service.addAddressUserAnswers(addressLookupState = addressLookupState,
+        address = AlfResponse(
+                                Some(organisation),
+                                List(addressLine1, addressLine2, addressLine3, addressLine4),
+                                Some(postcode),
+                                Some(countryCode)
+                              ),
+        userAnswers = emptyUserAnswers.copy(warehouseList = warehouseMap))
+
+      res.warehouseList mustBe Map("1" -> Warehouse(Some("super cola"),
+                                          UkAddress(List("33 Rhes Priordy", "East London"), "E73 2RP")),
+                                   "12" -> Warehouse(Some(organisation),
+                                          UkAddress(List(addressLine1, addressLine2, addressLine3, addressLine4), postcode)))
+    }
+
+    "add to the cache the address of a packing site when a user returns from address lookup frontend with missing address lines" in {
+      val addressLookupState = Packingdetails
+      val packingMap = Map("1" -> Site(UkAddress(List("33 Rhes Priordy", "East London"), "E73 2RP"),Some("ref1"), Some("super cola"),None))
+      val AddedPackingSiteMissingLines = Map("1" -> Site(UkAddress(List("33 Rhes Priordy", "East London"), "E73 2RP"),Some("ref1"), Some("super cola"),None),"12" -> Site(UkAddress(List(addressLine1, addressLine2), postcode),None, Some(organisation),None))
```

Comment:
same as above, can the packingMap contain a different id, so we can see it being added

Reply:
12 is the id of the one being added, it's just appended to the map so the initial one is still first